contracts/connection-router/src/state.rs

```diff
@@ -81,26 +82,61 @@ impl<'a> IndexList<ChainEndpoint> for ChainEndpointIndexes<'a> {
     }
 }
 
-// Message represents a message for which the fields have been successfully validated.
-// This should never be supplied by the user.
 #[cw_serde]
 pub struct NewMessage {
-    pub id_on_chain: MessageID, // globally unique
-    pub destination_address: Address,
+    pub id_on_chain: MessageID, // unique per chain
+    pub destination_address: ContractAddress,
```

Comment:
I almost wanna call this `chain_unique_id` or `cuid`, and the other one `cross_chain_unique_id` or `ccuid`.

Reply:
I like `chain_unique_id`. I wouldn't go with acronyms when it's not clear what they stand for 😆

Reply:
I would actually call this `source_chain_id` or something with source_chain, to make it clear this is an id from the source chain